src/main/scala/org/constellation/snapshotstreaming/FileBasedLastFullGlobalSnapshotStorage.scala

```diff
@@ -0,0 +1,56 @@
+package org.constellation.snapshotstreaming
+
+import cats.Applicative
+import cats.effect.Async
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.option._
+
+import org.tessellation.schema.GlobalSnapshot
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+import fs2.io.file.{Path, _}
+import fs2.{Stream, text}
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.parser.decode
+import io.circe.syntax._
+
+trait FileBasedLastFullGlobalSnapshotStorage[F[_]] {
+  def get: F[Option[Signed[GlobalSnapshot]]]
+}
+
+object FileBasedLastFullGlobalSnapshotStorage {
+
+  def make[F[_]: Async](path: Path) = new FileBasedLastFullGlobalSnapshotStorage[F] {
+    implicit val codec: Codec[Hashed[GlobalSnapshot]] = deriveCodec[Hashed[GlobalSnapshot]]
+
+    def set(snapshot: Hashed[GlobalSnapshot]): F[Unit] =
```

Comment:
This should be a part of the interface. It still works and can be called right now because `make` method has a complex derived type of:
```
FileBasedLastFullGlobalSnapshotStorage[IO] with Object {
  val codec : Codec[Hashed[GlobalSnapshot]]
  def set(snapshot: Hashed[GlobalSnapshot]): IO[Unit]
}
```
I would also recommend to declare the return type for `make`.